cmake/DAGMC_macros.cmake

```diff
@@ -0,0 +1,271 @@
+# All DAGMC libraries
+set(DAGMC_LIBRARY_LIST dagmc pyne_dagmc uwuw dagtally makeWatertight dagsolid fludag)
+
+macro (dagmc_setup_build)
+  message("")
+
+  # Set DAGMC version
+  set(DAGMC_MAJOR_VERSION 3)
+  set(DAGMC_MINOR_VERSION 0)
+  set(DAGMC_PATCH_VERSION 0)
+  set(DAGMC_VERSION ${DAGMC_MAJOR_VERSION}.${DAGMC_MINOR_VERSION}.${DAGMC_PATCH_VERSION})
+
+  # Default to a release build
+  if (NOT CMAKE_BUILD_TYPE)
+    message(STATUS "CMAKE_BUILD_TYPE not specified, defaulting to Release")
+    set(CMAKE_BUILD_TYPE Release)
+  endif ()
+  if (NOT CMAKE_BUILD_TYPE STREQUAL "Release" AND
+      NOT CMAKE_BUILD_TYPE STREQUAL "Debug" AND
+      NOT CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+    message(FATAL_ERROR "Specified CMAKE_BUILD_TYPE is invalid; valid options are Release, Debug, RelWithDebInfo")
+  endif ()
+  string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
+  message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+
+  # Installation directories
+  set(INSTALL_BIN_DIR     bin)
+  set(INSTALL_LIB_DIR     lib)
+  set(INSTALL_INCLUDE_DIR include)
+  set(INSTALL_TESTS_DIR   tests)
+  set(INSTALL_TOOLS_DIR   tools)
+  set(INSTALL_SHARE_DIR   share)
+
+  # Get some environment variables
+  set(ENV_USER "$ENV{USER}")
+  execute_process(COMMAND hostname       OUTPUT_VARIABLE ENV_HOST OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND uname -s       OUTPUT_VARIABLE ENV_OS   OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND date +%m/%d/%y OUTPUT_VARIABLE ENV_DATE OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND date +%H:%M:%S OUTPUT_VARIABLE ENV_TIME OUTPUT_STRIP_TRAILING_WHITESPACE)
+endmacro ()
+
+macro (dagmc_setup_options)
+  message("")
+
+  option(BUILD_MCNP5       "Build DAG-MCNP5"                         OFF)
```

Comment:
add a check for `MOAB_DIR` set here. (and maybe any other options which are set to `ON` and require a directory specification?

Reply:
I added a more verbose error message in `FindMOAB.cmake` if it can't find MOAB telling users how to specify it